### ISSUE TITLE
Priority conflict handling

### ISSUE THREAD

**Issue by droqen** [AmbientRun/Ambient#815]
More general case of #602 See detailed notes on Notion https://ambientrun.notion.site/Priority-Conflicts-8f2a226bc73b401787ad425ce38e7ca5?pvs=4

> In certain cases, for proper functioning we must assume there is one coherent input or output or organizing logic, even though we cannot avoid any number of entities from attempting internally consistent but collectively incoherent control/access

A very quickly proposed solution from that Notion doc would present the user with a per-conflict dialogue of some sort:

> There are 2 packages competing for exclusive write access to client_camera channel
> [view client_camera channel definition in package my_game]
> - fps_controller [view source] [grant exclusive access]
> - client_my_game [view source] [grant exclusive access]

**Comment by droqen** [comment]
Feel free to close or move this for being more of a 'discussion' than an 'issue'; I recognize it's composed of _multiple_ dream pieces that don't even exist yet. A dreamy "ideal" version of how this would work.

**Comment by FredrikNoren** [comment]
Hm I don't think these things should be thought of together. Camera, UI focus, enemy ai and player spawning are very different and likely require different solutions particular to each of them.

**Comment by droqen** [comment]
I agree they're different cases that will need particular solutions, but we're missing even a primitive tool or concept for defining this type of relationship at a package level -- if the particular solutions we employ are always _engine-only_ (e.g. `camera::set_global_camera()`) then users won't be able to properly define their own solutions when new such cases arise.

When I encounter individual cases I'll be sure to jot down some notes on them.

**Comment by droqen** [comment]
One example given in my recent notes on integrating AFPS

https://ambientrun.notion.site/Integrating-AFPS-into-a-new-package-b202f51d53244b3bb81e9c1b64d951c6?pvs=25#e5786c5c0f3d412f83f5eab9c3605a81

> For replacing AI logic, I can define a component that is required for a zombie to follow the player, and then replace that component. [.. But,] There is not currently a way to prevent multiple [AI logic] systems from ‘conflicting’, i.e. acting upon the same entity/components in ways that produce incoherent behaviour.
